projects/calc/js/main.js

```diff
@@ -0,0 +1,365 @@
+"use strict";
+
+class Calc {
+  static get MAX_NUMBER() {
+    return 24;
+  }
+  constructor(options) {
+    this.elem = options.elem;
+
+    this._operand1 = '';
+    this._operand2 = '';
+    this._operator = null;
+    this._activeOperand = '';
+    this._result = '';
+    this._hasError = false;
+
+    this._resultField = document.querySelector('#result-field');
+
+    // All available operators
+    this._allOperators = {
+      '+': function(obj) {
+        return obj.a + obj.b;
+      },
+      '-': function(obj) {
+        return obj.a - obj.b;
+      },
+      '*': function(obj) {
+        return obj.a * obj.b;
+      },
+      '/': function(obj) {
+        return obj.a / obj.b;
+      },
+      '1/x': function(obj) {
+        return 1 / obj.a;
+      },
+      'sqrt': function(obj) {
+        return Math.sqrt(obj.a);
+      },
+      '%': function(obj) {
+        return obj.a * obj.b / 100;
+      },
+      'pow2': function(obj) {
+        return Math.pow(obj.a, 2);
+      },
+      'pow3': function(obj) {
+        return Math.pow(obj.a, 3);
+      },
+      'powy': function(obj) {
+        return Math.pow(obj.a, obj.b);
+      },
+      '10pow': function(obj) {
+        return Math.pow(10, obj.a);
+      },
+      'factorial': function(obj) {
+        function factorial(n) {
+          return n ? n * factorial(n - 1) : 1;
+        }
+
+        return factorial(obj.a);
+      }
+    }
+
+    // Buttons event listener
+    this.elem.addEventListener('click', function(e) {
+      var target = e.target;
+
+      if(target.tagName === 'A') {
+        if(!this._hasError || (this._hasError && target.dataset.type === 'clear')) {
```

Comment:
there is no need to check again for `this._hasError`
after `||` it is always `true`